### PR TITLE
prefer 'agg' for non-interactive R

### PR DESCRIPTION
**Pull request by kevinushey**
Addresses https://github.com/rstudio/rstudio/issues/13840.

@cderv does this look like the right fix to you?

### CODE REVIEW

File: R/knitr-engine.R [hunk 1]
```diff
@@ -492,19 +492,44 @@ eng_python_initialize_hooks <- function(options, envir) {
...
+  for (envvar in envvars) {
+    override <- Sys.getenv(envvar, unset = NA)
+    if (!is.na(override))
+      return(override)
+  }
+
+  # in RStudio Desktop, enforce a non-Qt matplotlib backend
+  # this is especially important with RStudio Desktop as attempting to use a Qt
+  # backend will cause issues due to mismatched Qt versions between RStudio and
+  # Anaconda environments, and will cause crashes when generating plots
```
Comment (t-kalinowski) [inline R/knitr-engine.R:508]: Is this a conda specific workaround? Would a venv python installation find the same Qt version as the IDE? If not, should the comment be updated?

File: R/knitr-engine.R [hunk 2]
```diff
@@ -492,19 +492,44 @@ eng_python_initialize_hooks <- function(options, envir) {
...
   # early exit if we already initialized
   # (this onload hook is registered for multiple matplotlib submodules)
   if (identical(.globals$matplotlib_initialized, TRUE))
     return(TRUE)
 
-  # attempt to enforce a non-Qt matplotlib backend. this is especially important
-  # with RStudio Desktop as attempting to use a Qt backend will cause issues due
-  # to mismatched Qt versions between RStudio and Anaconda environments, and
-  # will cause crashes when attempting to generate plots
-  testthat <- Sys.getenv("TESTTHAT", unset = NA)
```
Comment (t-kalinowski) [inline R/knitr-engine.R:506]: Should we keep the check for if we're in a testthat context? I think we can run tests interactively too.

**Review by t-kalinowski — APPROVED** [review]
LGTM overall!

File: R/knitr-engine.R (hunk 2 above)
Comment (kevinushey) [reply R/knitr-engine.R:506]: Thanks for catching that -- I've brought that back.

File: R/knitr-engine.R (hunk 1 above)
Comment (kevinushey) [reply R/knitr-engine.R]: IIRC, this branch was specifically required for older versions of RStudio (which were built against Qt); I've updated the comment to indicate that and also note that we could probably remove this in a future `reticulate` release. Does that seem sensible?

**Comment by cderv** [comment]
Sorry I missed this on Monday. Looks good thanks !

**Pull request merged**